File: translations/codenet/deepseek_commented/C++2P/granite/Code/s605014348.py

import math

# Function to find the smallest divisor of a number
def div(n):
    if n % 2 == 0:
        return 2
    s = math.sqrt(n)
    for i in range(3, int(s) + 1, 2):
        if n % i == 0:
            return i
    return n

num = int(input("Enter an integer: "))
print(num, ":", end=" ")

while num!= 1:
    print(div(num), end=" ")
    num = num / div(num)

print()

File: translations/codenet/deepseek_commented/C++2Go/granite/Code/s605014348.go

package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
)

// Function to find the smallest divisor of a number
func div(n int) int {
	if n%2 == 0 {
		return 2
	}
	s := int(math.Sqrt(float64(n)))
	for i := 3; i <= s; i += 2 {
		if n%i == 0 {
			return i
		}
	}
	return n
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	fmt.Print("Enter an integer: ")
	text, _ := reader.ReadString('\n')
	num, _ := strconv.Atoi(text)
	fmt.Print(num, ":")
	for num!= 1 {
		fmt.Print(" ", div(num))
		num = num / div(num)
	}
	fmt.Println()
}

// 